src/fomo.jl

```diff
@@ -0,0 +1,307 @@
+export fomo, FomoSolver, tr, qr
+
+abstract type AbstractFomoMethod end
+
+struct tr <: AbstractFomoMethod end
+struct qr <: AbstractFomoMethod end
```

Comment:
Could this be called `r2`? It would help with the transition and avoid a conflict with the QR factorization.